Explorer/Assets/Scripts/Global/Dynamic/PortableExperiences/PortableExperiencesController.cs

```diff
@@ -0,0 +1,218 @@
+                return portableExperienceComponent.ParentSceneId == currentSceneFacade.Info.Name;
+            }
+
+            return false;
+        }
+
+        public List<IPortableExperiencesApi.SpawnResponse> GetAllPortableExperiences()
+        {
+            spawnResponsesList.Clear();
+
+            foreach (var portableExperience in PortableExperienceEntities)
+            {
+                PortableExperienceComponent pxComponent = world.Get<PortableExperienceComponent>(portableExperience.Value);
+
+                spawnResponsesList.Add(new IPortableExperiencesApi.SpawnResponse {
+                        name = pxComponent.RealmData.RealmName,
+                        ens = portableExperience.Key,
+                        parent_cid = pxComponent.ParentSceneId,
+                        pid = portableExperience.Value.Id.ToString() });
+            }
+
+            return spawnResponsesList;
+        }
+
+        public async UniTask<IPortableExperiencesApi.ExitResponse> UnloadPortableExperienceAsync(string ens, CancellationToken ct)
+        {
+            if (!EnsUtils.ValidateEns(ens)) throw new ArgumentException($"The provided ens {ens} is invalid");
+
+            //TODO: We need to be able to kill PX using only the urn as well, it will mean some changes to this code, this will be done in the next iteration.
+
+            await UniTask.SwitchToMainThread();
+
+            //We need to dispose the scene that the PX has created.
+            if (PortableExperienceEntities.TryGetValue(ens, out Entity portableExperienceEntity))
+            {
+                PortableExperienceComponent portableExperienceComponent = world.Get<PortableExperienceComponent>(portableExperienceEntity);
+
+                //Portable Experiences only have one scene
+                string? sceneUrn = portableExperienceComponent.Ipfs.SceneUrns[0];
+                string sceneEntityId = string.Empty;
+
+                if (sceneUrn != null)
+                {
+                    sceneEntityId = IpfsHelper.ParseUrn(sceneUrn).EntityId;
+
+                    if (scenesCache.TryGetPortableExperienceBySceneUrn(sceneEntityId, out ISceneFacade sceneFacade))
+                    {
+                        await sceneFacade.DisposeAsync();
+                        scenesCache.RemovePortableExperienceFacade(sceneEntityId);
+                    }
+                }
+
+                // Clear the world from everything connected to the current PX
+                //for this we will need to go over all these entities in the query
+                //and check if their IpfsPath.EntityId coincides with the scene's entityId and if so, delete them.
+                entitiesToDestroy.Clear();
+
+                if (!string.IsNullOrEmpty(sceneEntityId))
+                {
+                    GetEntitiesToDestroy(sceneEntityId, GET_SCENE_DEFINITION, CheckGetSceneDefinitions, ref entitiesToDestroy);
```

Comment:
It contradicts the general flow of destruction: it gives no opportunity for systems to react on `DeleteEntityIntention`. Here the loading flow won't be properly cancelled because `AssetPromise.ForgetLoading` should be called. 

I propose you to arrange everything in the systems properly by placing `DeleteEntityIntention`, handle it accordingly for PX. (Please keep in mind that for normal scenes `EntityDefinitionComponent` is not destroyed)